Add USWDS Government Banner

kmonahan opened the pull request:
Adds the USWDS government banner (usa-banner) and the media block (usa-media-block) components, because the banner uses media block as part of its markup. Also updated the method of pulling SCSS in to remove the individual imports and instead import packages/uswds-components in the main sass file. Since USWDS components may depend on other components (e.g. the government banner uses both the accordion and the media block), importing the packages file ensures that any dependencies gets included.

kmonahan committed:
Add USWDS Government Banner

dcmouyard commented on source/styles.scss:
```diff
@@ -23,7 +23,7 @@
 
 // USWDS components.
-@import '_patterns/03-uswds/**/*.scss';
+@import 'packages/uswds-components';
```
I like the simplicity of including everything, and removing Sass partials from the USWDS section of PL makes it more obvious that those styles shouldn’t be changed.

That said, it does obfuscate where the USWDS styles are coming from with no hint in the Sass partial pointing to the specific components in USWDS we’re importing.

Also, I can imagine smaller budget projects would want to quickly override the USWDS components and not want to spend a bunch of time creating a new variant in the Components section. Should we really enforce that USWDS components shouldn’t be quickly overridden in the USWDS section?

I’m also a little worried with all of the extra CSS we’re generating by including _all_ of the USWDS components, even those we’re not using on a project.

kmonahan replied:
@dcmouyard,
As far as obfuscation, I think we should continue the pattern I started in the earlier components (and then dropped off a bit, so I can go back and update) of including the link to the USWDS documentation about the pattern in the markdown file.

Based on our earlier conversations, I thought the intention of the USWDS component directory was to demo the patterns without modifications, in order to allow the USWDS version to be updated in the future, and that any modifications should be done by extending or overriding components in the components directory. If a potential future use case we want to support _is_ modifying patterns directly in the USWDS directory, then that changes my preferred approach here.

I'm not particularly worried about the extra CSS, because we're including a lot of the USWDS components already. A slightly larger CSS file but a developer not spending time debugging why the government banner isn't working because they didn't include the accordion component is a trade-off I'm willing to make, unless we do want to change our plan a bit and support overriding components directly the the uswds directory.

dcmouyard replied:
> Based on our earlier conversations, I thought the intention of the USWDS component directory was to demo the patterns without modifications, in order to allow the USWDS version to be updated in the future, 

This is all still true for this fork.

> and that any modifications should be done by extending or overriding components in the components directory.

This part was only about the EPA project since they care about easily updating USWDS in the future. I’m not sure that will be the case on most other projects. That said, I could easily be persuaded in either direction as long as we document it.

> I'm not particularly worried about the extra CSS, because we're including a lot of the USWDS components already. A slightly larger CSS file but a developer not spending time debugging why the government banner isn't working because they didn't include the accordion component is a trade-off I'm willing to make, unless we do want to change our plan a bit and support overriding components directly the the uswds directory.

Sounds good to me.

dcmouyard reviewed (commented):
I’m leaning towards including the banner by default at the top of the default page template. `_patterns/07-pages/_page-default.twig`

@kmonahan  How do you feel about that?

kmonahan committed:
Add back separate SCSS files and add dependency listing

kmonahan committed:
Add government banner to _page-default

kmonahan committed:
Fix typo in filename

kmonahan committed:
Add back additional scss

kmonahan committed:
Merge branch '8.x-3.x' into banner-component

kmonahan committed:
Import scss from USWDS directory

kmonahan commented:
Updated based on our conversation:

- Removed the packages/components import and switched back to individual scss files.
- Listed the dependencies in the markdown files where relevant.
- Added the government banner to _page-default.twig

dcmouyard reviewed (approved):
👍
